src/main/kotlin/io/slama/commands/call_command.kt

```diff
@@ -7,83 +7,101 @@ import io.slama.utils.TaskScheduler
-            event.getOption("role")?.asRole
+            event.jda,
+            event.member!!,
+            event.channel.asTextChannel(),
+            event.getOption("timeout")?.asLong ?: DEFAULT_CALL_TIMEOUT,
+            event.getOption("role")?.asRole,
+            event
         )
     }
 }
 
-private class Call(
-    private val event: SlashCommandInteractionEvent,
-    private val timeout: Long = DEFAULT_TIMEOUT,
-    private val role: Role? = null
+class Call(
+    private val jda: JDA,
+    private val member: Member,
+    private val channel: TextChannel,
+    private val timeout: Long = DEFAULT_CALL_TIMEOUT,
+    private val role: Role? = null,
+    event: SlashCommandInteractionEvent? = null,
 ) : ListenerAdapter() {
     private val students = mutableSetOf<String>()
     private val missing = mutableSetOf<String>()
-    private val uniqueId = "${event.user.id}${System.currentTimeMillis()}"
-    private val embedTitle = "Appel demandé par ${event.member?.effectiveName ?: "un certain A. N. Onym"}"
+    private val uniqueId = "${member.user.id}${System.currentTimeMillis()}"
+    private val embedTitle = "Appel demandé par ${member.effectiveName}"
 
     private lateinit var responseId: String
     private var totalTargeted: Int = 0
 
     init {
-        event.jda.addEventListener(this)
-        logger.info("${event.member} initiated a call in ${event.channel} with a timeout of $timeout minutes")
+        jda.addEventListener(this)
+        logger.info("$member initiated a call in $channel with a timeout of $timeout minutes")
 
         role?.guild?.findMembers { role in it.roles }?.onSuccess { members ->
             members.map { it.effectiveName }.forEach { missing.add(it) }
             totalTargeted = members.size
         }
 
-        event.replyEmbeds(
-            EmbedBuilder()
-                .setTitle(embedTitle)
-                .setDescription("Vous avez $timeout ${"minute".pluralize(timeout.toInt())} pour répondre à l'appel.")
-                .setColor(EmbedColors.GREEN)
-                .build()
-        )
-            .addActionRow(Button.success("$uniqueId.respond", "Répondre à l'appel"))
-            .queue {
-                it.retrieveOriginal().queue { message -> responseId = message.id }
-                TaskScheduler.later(timeout, TimeUnit.MINUTES, ::sendResult)
-            }
+        val embed = EmbedBuilder()
+            .setTitle(embedTitle)
+            .setDescription("Vous avez $timeout ${"minute".pluralize(timeout.toInt())} pour répondre à l'appel.")
+            .setColor(EmbedColors.GREEN)
+            .build()
+
+        if (event != null) {
+            event.replyEmbeds(embed)
+                .addActionRow(Button.success("$uniqueId.respond", "Répondre à l'appel"))
+                .queue {
+                    it.retrieveOriginal().queue { message -> responseId = message.id }
+                    TaskScheduler.later(timeout, TimeUnit.MINUTES, ::sendResult)

```

Comment:
```suggestion
                    it.retrieveOriginal().queue { message -> 
                        responseId = message.id 
                        TaskScheduler.later(timeout, TimeUnit.MINUTES, ::sendResult)
                    }

```

shouldn't this be queued like that?, to mirror the event == null case

Reply:
Well here it is different because we are getting the `responseId` from an event, and because of Discord cache erasure we're forced to fetch. This has nothing to do with the coroutine because if this fetch fails it should not prevent the call from ending.